### PR TITLE
Initial payments page design

### PR DESCRIPTION
**Pull request by corymcdonald**
I did quite a few stuff in this PR here are some main overviews

Part one of this issue #1531

* Add prettier
	It was recommended to just use prettier which is an opinionated JS framework so we don't have to argue about styles and stuff. https://twitter.com/dan_abramov/status/1086215004808978434

* Convert `locale.js` from javascript to typescript
* Rename most files from `index.tsx` to be the component name

screen shot 2019-01-24 at 1 29 35 pm
screen shot 2019-01-24 at 1 28 20 pm

### CODE REVIEW

File: app/controllers/payments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class PaymentsController < ApplicationController
```
Comment (corymcdonald) [inline app/controllers/payments_controller.rb]: Going to restrict controller access to the new ui similar to 
https://github.com/brave-intl/publishers/pull/1495/files#diff-35a55926102be0f7565d1bea8f1207e5R374

**Comment by dlipeles** [comment]
Just a cursory review, will take a deeper dive on Monday , looks like a good start.

File: app/javascript/views/payments/Payments.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import * as React from "react";
+
+import { Container, Wrapper } from "../style";
+import { Card, Layout, Row } from "./style";
```
[the remark this answers is not in the document]
Comment (corymcdonald) [reply app/javascript/views/payments/Payments.tsx]: Let's switch this to `componentNameStyle`

File: config/routes.rb [hunk 3]
```diff
@@ -39,6 +39,7 @@
       resources :totp_registrations, only: %i(new create destroy)
       resources :totp_authentications, only: %i(create)
       resources :promo_registrations, only: %i(index create)
+      resources :payments
```
Comment (dlipeles) [inline config/routes.rb]: Okay got a lot of food for thought on using resources here, let me dm you on slack.

File: app/javascript/views/payments/paymentHistory/PaymentHistory.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import * as React from "react";
+
+import locale from "../../../locale/en";
+import downloadIcon from "./downloadIcon.png";
```
[the remark this answers is not in the document]
Comment (corymcdonald) [reply app/javascript/views/payments/paymentHistory/PaymentHistory.tsx]: It's actually already an icon!

File: app/javascript/views/payments/paymentHistory/PaymentHistory.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+              <Cell>999.9 EURO</Cell>
+              <Cell>
+                <Link href="#">View</Link>
+                <Link href="#">
+                  <img
+                    src={downloadIcon}
+                    style={{ marginRight: "5px" }}
+                    width={22}
+                    height={19}
+                  />
```
[the remark this answers is not in the document]
Comment (corymcdonald) [reply app/javascript/views/payments/paymentHistory/PaymentHistory.tsx]: Switched to a downloadIcon

File: app/javascript/views/payments/upholdCard/UpholdCard.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
+import * as React from "react";
+
+import { CaratRightIcon } from "brave-ui/components/icons";
+import locale from "../../../locale/en";
```
[the remark this answers is not in the document]
Comment (corymcdonald) [reply app/javascript/views/payments/upholdCard/UpholdCard.tsx]: https://github.com/brave-intl/publishers/pull/1555/commits/886a4d60d3f1eea48ce955bdfd09b5936da2ffef

done!

File: app/controllers/payments_controller.rb (hunk 1 above)
Comment (corymcdonald) [reply app/controllers/payments_controller.rb]: https://github.com/brave-intl/publishers/pull/1555/commits/886a4d60d3f1eea48ce955bdfd09b5936da2ffef

File: config/routes.rb (hunk 3 above)
Comment (corymcdonald) [reply config/routes.rb]: Dan and I discussed and I think we have some ideas on what needs to happen here 👍

File: app/javascript/views/payments/Payments.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (corymcdonald) [reply app/javascript/views/payments/Payments.tsx]: https://github.com/brave-intl/publishers/pull/1555/commits/7438b6bd2f6933115ff5451347711135e1e6ffac

**Pull request merged**